### PR TITLE
Add and modify commands' README files

### PR DESCRIPTION
**Pull request by blackdwarf**
Update those READMEs that are outdated. Add READMEs for dotnet pack
and dotnet restore as well.

skip ci please

As @leecow said, docs-only PRs do not need shiproom approval.

/cc @piotrpMSFT @brthor @Sridhar-MS @livarcocc @anurse

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2338)

### CODE REVIEW

File: src/dotnet/commands/dotnet-build/README.md [hunk 1]
```diff
@@ -1,38 +1,87 @@
-dotnet-build
-===========
+% DOTNET-BUILD(1)
+% Zlatko Knezevic zlakne@microsoft.com
```
Comment (krwq) [inline src/dotnet/commands/dotnet-build/README.md]: I think we should get rid of any personal info

Comment (brthor) [inline src/dotnet/commands/dotnet-build/README.md]: should we have a more general email here?

Comment (leecow) [inline src/dotnet/commands/dotnet-build/README.md]: Agreed on stripping personal info.

Comment (blackdwarf) [inline src/dotnet/commands/dotnet-build/README.md]: We actually did this because man pages usually have these. Remove the email, author or both?

Comment (anurse) [inline src/dotnet/commands/dotnet-build/README.md]: Change the author to "Microsoft Corporation" and use a feedback DL for the email address?

Comment (blackdwarf) [inline src/dotnet/commands/dotnet-build/README.md]: Sure. We don't have the feedback DL though. Maybe I should make one.

**Comment by blackdwarf** [comment]
Is this OK now? Can I merge it?

**Comment by blackdwarf** [comment]
OK, merging this. Just MD files anyway.

**Pull request merged**